File: src/AccelByte.Extend.Vivox.Authentication.Server/Vivox/TokenClaims.cs
// Copyright (c) 2024 AccelByte Inc. All Rights Reserved.
// This is licensed software from AccelByte Inc, for limitations
// and restrictions contact your company contract manager.

using System;
using System.Text.Json.Serialization;

namespace AccelByte.Extend.Vivox.Authentication.Server.Vivox
{
    public class TokenClaims
    {
        [JsonPropertyOrder(1)]
        [JsonPropertyName("vxi")]
        public int UniqueId { get; set; } = 1;

        [JsonPropertyOrder(2)]
        [JsonPropertyName("sub")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Subject { get; set; } = null;

        [JsonPropertyOrder(3)]
        [JsonPropertyName("f")]
        public string From { get; set; } = "";

        [JsonPropertyOrder(4)]
        [JsonPropertyName("iss")]
        public string Issuer { get; set; } = "";

        [JsonPropertyOrder(5)]
        [JsonPropertyName("vxa")]
        public string Action { get; set; } = "";

        [JsonPropertyOrder(6)]
        [JsonPropertyName("t")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? To { get; set; } = null;

        [JsonPropertyOrder(7)]
        [JsonPropertyName("exp")]
        public long ExpiryTime { get; set; } = 0;
    }
}

File: src/main/java/net/accelbyte/extend/serviceextension/vivox/TokenClaims.java
package net.accelbyte.extend.serviceextension.vivox;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.Data;

@Data
@JsonPropertyOrder({ "vxi", "sub", "f", "iss", "vxa", "t", "exp" })
public class TokenClaims {

    @JsonProperty("vxi")
    private int uniqueId = 1;

    @JsonProperty("sub")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String subject = null;

    @JsonProperty("f")
    private String from = "";

    @JsonProperty("iss")
    private String issuer = "";

    @JsonProperty("vxa")
    private String action = "";

    @JsonProperty("t")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String to = null;

    @JsonProperty("exp")
    private long expiryTime = 0;
}